components/zoho_crm/actions/add-attachment/add-attachment.js

```diff
@@ -0,0 +1,94 @@
+const common = require("../common");
+const { zoho_crm } = common.props;
+const validate = require("validate.js");
+
+module.exports = {
+  key: "zoho_crm-add-attachment",
+  name: "Add Attachment",
+  description: "Adds a file attachment to the given module record.",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    zoho_crm,
+    domainLocation: {
+      propDefinition: [
+        zoho_crm,
+        "domainLocation",
+      ],
+    },
+    module: {
+      type: "string",
+      label: "Module",
+      description:
+        "The module related to the record you'd like to add the attachment.",
+      options: [
+        "Leads",
+        "Accounts",
+        "Contacts",
+        "Deals",
+        "Campaigns",
+        "Tasks",
+        "Cases",
+        "Events",
+        "Calls",
+        "Solutions",
+        "Products",
+        "Vendors",
+        "Price_Books",
+        "Quotes",
+        "Sales_Orders",
+        "Purchase_Orders",
+        "Invoices",
+        "Custom",
+        "Notes",
+      ],
```

Comment:
I checked the values return by the related API endpoint to that method, I am getting:
```
Home
Leads
Contacts
Accounts
Deals
Activities
Reports
Feeds
Campaigns
Documents
Visits
Tasks
Events
Notes
Attachments
Calls
Actions_Performed
```
but we actually need:
```
      options: [
        "Leads",
        "Accounts",
        "Contacts",
        "Deals",
        "Campaigns",
        "Tasks",
        "Cases",
        "Events",
        "Calls",
        "Solutions",
        "Products",
        "Vendors",
        "Price_Books",
        "Quotes",
        "Sales_Orders",
        "Purchase_Orders",
        "Invoices",
        "Custom",
        "Notes",
      ],
```
so, should leave the hardcoded options as it is now, or do a compare againstg the resultset of `listModules` and just add any modules missing. As you can see modules such as `Purchase_Orders` or `Sales_Orders` are not in the result set.

the elegible modules for adding a tag (the related actions) are in the [Add tags to multiple records](https://www.zoho.com/crm/developer/docs/api/v2/add-tags.html) Api docs.

Reply:
That's because of your particular account settings. The list of available modules will depend on the account settings.
That's why we need to dynamically retrieve this list, because it's not going to be the same for everyone.

Reply:
Ok, options are dynacamilly poplated from `listModules`